core/src/main/java/io/paradoxical/cassieq/dataAccess/MessageRepositoryImpl.java

```diff
@@ -261,4 +265,26 @@ private void updateQueueSize(int amount){
 
         session.execute(with);
     }
+
+    /**
+     * Determines if the message is allowed to be consumed
+     * @param message
+     * @return
+     */
+    private boolean messageConsumable(final Message message) {
+        if(message.getDeliveryCount() >= queueDefinition.getMaxDeliveryCount()){
```

Comment:
Totally agree. I chose to put it here since consume is called from a bunch of spots and I was concerned about having to remember to tool in this logic in both invis and reader.

open to suggestions